pkg/features/features.go

```diff
@@ -39,7 +40,16 @@ func NewFeatureGateOptionsOrDie(featureGates featuregate.MutableFeatureGate, pro
 
 func (o *FeatureGateOptions) AddFlags(cmd *cobra.Command) {
 	flags := cmd.Flags()
-	flags.Var(flag.NewMapStringBool(&o.featureGateArgs), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+	flags.Var(k8sflag.NewMapStringBool(&o.featureGateArgs), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+		"Options are:\n"+strings.Join(o.featureGates.KnownFeatures(), "\n"))
+}
+
+func (o *FeatureGateOptions) AddFlagsToGoFlagSet(flagset *flag.FlagSet) {
+	if flagset == nil {
+		flagset = flag.CommandLine
+	}
```

Comment:
Are you sure this is going to work? 

This is just changing the local copy of `flagset`, the original variable will continue to be nil.

Reply:
`flag.CommandLine` is a global variable, so by assigning here, we are effectively calling `flag.CommandLine.Var` on line 52, which is the intended use to add global flags, so I'm pretty sure this is ok

Reply:
Ah yes, I thought the same at first as well! It's not clear that `flag.CommandLine` is a global var :) It's the same thing that `klog` does when you do a `klog.InitFlags(nil)`

I've also verified it works in a local build.

Reply:
Ack, that makes sense. Sorry for the noise.
/hold cancel

Reply:
No worries! Do you know who I should ping for approval? :)

Reply:
CC @soltysh